### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by syrenaaa** [bigratmonster/bigrat.monster#28] [General]
ex. choose a random image from the monkeys/ directory
or something lol

**Comment by syrenaaa** [comment]
oh this is the first issue? nice

**Comment by humboldt123** [comment]
That sounds great, why don't you make a pr :sunglasses:

**Comment by SoNothingMC** [comment]
There's a reason why this repo has no "feature request" issue templates.

Because I do not have enough programming knowledge to implement stuffs like this 😥

**Comment by humboldt123** [comment]
Ill do it

**Comment by SoNothingMC** [comment]
pogchamp

**Comment by humboldt123** [comment]
how 2 build A P I?????

**Comment by smokeytube** [comment]
is this a thing yet

**Comment by d1snin** [comment]
> Ill do it

use ktor

**Comment by d1snin** [comment]
https://github.com/ktorio/ktor

**Comment by smokeytube** [comment]
my tiny brain cant handle non-scripting languages

**Comment by syrenaaa** [comment]
Nep nep why did u change name

**Comment by d1snin** [comment]
> Nep nep why did u change name

:trollface:

**Comment by smokeytube** [comment]
so api when

**Comment by EmeraldSnorlax** [comment]
why is this even needed for static images
what would an api for this even do
and if there is an api, id assume it'd be serverless

**Comment by smokeytube** [comment]
idc what the api does just make it

**Comment by SoNothingMC** [comment]
#29

**Comment by syrenaaa** [comment]
sus

**Comment by EmeraldSnorlax** [comment]
im this close to making discussions and issues contributor only :trollface:

**Comment by SoNothingMC** [comment]
api when

**Comment by d1snin** [comment]
when the api

**Comment by smokeytube** [comment]
still waiting

**Comment by SoNothingMC** [comment]
how

**Comment by d1snin** [comment]
so true

**Comment by d1snin** [comment]
review and merge my pr rn

**Comment by EmeraldSnorlax** [comment]
weve somehow ended up with two apis both with issues that need to be resolved, so i have no idea which to go with

the js api needs a significant refactor and then a port to cf workers, but the java api has nowhere to be hosted

im curious to everyones thoughts on such

**Comment by SoNothingMC** [comment]
here's my phone's thought:

![image](https://user-images.githubusercontent.com/32032824/126474149-4dd8cdb4-d172-4f0f-8c24-341fb05f22d0.png)

**Reply by EmeraldSnorlax** [reply]
crop your images holy shit lol

**Comment by syrenaaa** [comment]
:sob:

**Comment by d1snin** [comment]
Well, I have a great project idea, we can make a simple api that will return a random image of a rat (more precisely, a link to this image), these links will be stored in a database (I plan to use mongo) and the coolest thing is that users will be able to add their own links and their own image of rats, for this I will also make API and corresponding verifications, then this image will be sent for moderation to selected maintainers (probably this will be done through a discord bot) all this will be packaged by microservices and deployed via docker compose

**Comment by d1snin** [comment]
@SoNothingMC what do u think about it :trollface:

**Comment by d1snin** [comment]
due to the fact that users can add their own rats, we can use snorlax to make some kind of gui

**Comment by d1snin** [comment]
the api will also provide json responses

**Comment by syrenaaa** [comment]
sounds like a lot of work

**Reply by d1snin** [reply]
not really, I will do it alone in a month, even though I am busy learning often

**Comment by d1snin** [comment]
I'll just make a task list here

- [ ] Configure database support
- [ ] Introduce docker compose support
- [ ] Configure the API and support for adding new rats from users.
- [ ] configure the discord bot for content moderation
- [ ] set up also local storage and the use of cdn for content distribution (we certainly won't have many users, I'm just interested in working with this)

**Comment by EmeraldSnorlax** [comment]
https://github.com/bigratmonster/bigrat.monster/issues/35

thread resolved
